radicle-cli/src/commands/remote/list.rs

```diff
@@ -19,23 +17,20 @@ pub fn run(repo: &git::Repository) -> anyhow::Result<()> {
     for r in remotes {
         for spec in r.refspecs() {
             let dir = spec.direction();
-            let url = r.url().ok_or(anyhow::anyhow!("url not present"))?;
-            let name = r
-                .name()
-                .ok_or(anyhow::anyhow!("remote name is not present"))?;
+            let url = r.url.clone();
+            let name = r.name.clone();
             table.push([
                 term::format::badge_positive(format_direction(&dir)),
                 term::format::highlight(name.to_owned()),
-                term::format::highlight(url_to_nid(url)?),
+                term::format::highlight(url_to_nid(&url)?),
             ]);
         }
     }
     table.print();
     Ok(())
 }
 
-fn url_to_nid(url: &str) -> anyhow::Result<String> {
-    let url = radicle::git::Url::from_str(url)?;
+fn url_to_nid(url: &radicle::git::Url) -> anyhow::Result<String> {
```

Comment:
This function is probably not needed now, but maybe I'm wrong.

Reply:
No you are right, while I was working on a @cloudhead suggestion I found that I can use your idea and remote this to improve the code.

My solution was ugly, now should be better there https://github.com/radicle-dev/heartwood/pull/524/commits/dc02994ec857798ec6c70f97f3f0a254b9bf8669